src/rmarkdown/index.ts

```diff
@@ -60,17 +60,20 @@ export class RMarkdownCodeLensProvider implements vscode.CodeLensProvider {
         this.codeLenses = [];
         const chunks = getChunks(document);
         const chunkRanges: vscode.Range[] = [];
-        const rmdCodeLensCommands: string[] = config().get('rmarkdown.codeLensCommands');
+        const rmdCodeLensCommands: string[] = config().get('rmarkdown.codeLensCommands', []);
 
         // Iterate through all code chunks for getting chunk information for both CodeLens and chunk background color (set by `editor.setDecorations`)
         for (let i = 1; i <= chunks.length; i++) {
             const chunk = chunks.find(e => e.id === i);
+            if (!chunk) {
+                continue;
+            }
             const chunkRange = chunk.chunkRange;
             const line = chunk.startLine;
             chunkRanges.push(chunkRange);
 
             // Enable/disable only CodeLens, without affecting chunk background color.
-            if (config().get<boolean>('rmarkdown.enableCodeLens') && (chunk.language === 'r') || isRDocument(document)) {
+            if (config().get<boolean>('rmarkdown.enableCodeLens', false) && (chunk.language === 'r') || isRDocument(document)) {
```

Comment:
`r.rmarkdown.enableCodeLens` is by default  `true`.

Reply:
I'm curious and wonder if `config().get<T>()` could guarantee that the default value could be returned if somehow missing?

Reply:
I don't think its possible at compile / "type check" time. For correctness I think all occurrences of config().get<T>() should in the future (carefully) be replaced with the defaults. Maybe we could introduce a lint for that.

Reply:
Fixed in 5418748